web/ui/react-app/src/pages/graph/GraphControls.tsx

```diff
@@ -58,6 +58,17 @@ class GraphControls extends Component<GraphControlsProps> {
     this.rangeRef.current!.value = formatDuration(range);
   };
 
+  onEnterRangeInput = (e: React.KeyboardEvent<HTMLInputElement>): void => {
+    if (e.key === 'Enter') {
+      const range = parseDuration(this.rangeRef.current!.value);
```

Comment:
This function body is mostly a duplication of `onChangeRangeInput()`, how about just:

```typescript
  onEnterRangeInput = (e: React.KeyboardEvent<HTMLInputElement>): void => {
    if (e.key === 'Enter') {
      this.onChangeRangeInput(this.rangeRef.current!.value);
    }
  };
```

But since this function is not only called on enter, it is actually the general `KeyDown` handler, so we should probably either rename it to `onRangeInputKeyDown()` or similar, or just change the event handler below to this inline version:

```typescript
onKeyDown={(e: React.KeyboardEvent<HTMLInputElement>) => e.key === 'Enter' && this.onChangeRangeInput(this.rangeRef.current!.value)}
```

Then the PR becomes one line :)

Reply:
Sounds good. I was actually trying this one line approach at first but wasn't lucky with the `? :` ternary operator. Looks good above!